_overviews/scala3-book/ca-type-classes.md

```diff
@@ -20,99 +20,171 @@ A type class is useful in multiple use-cases, for example:
 
 ## Example
 
-For example, `Show` is a well-known type class in Haskell, and the following code shows one way to implement it in Scala 3.
-If you imagine that Scala classes don’t have a `toString` method, you can define a `Show` type class to add this behavior to any class that you want to be able to convert to a custom string.
+For example, `Show` is a well-known type class in Haskell, and the following code shows one way to implement it in Scala.
+If you imagine that Scala classes don’t have a `toString` method, you can define a `Show` type class to add this behavior to any type that you want to be able to convert to a custom string.
 
 ### The type class
 
 The first step in creating a type class is to declare a parameterized trait that has one or more abstract methods.
 Because `Showable` only has one method named `show`, it’s written like this:
 
+{% tabs 'definition' class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+```scala
+// a type class
+trait Showable[A] {
+  def show(a: A): String
+}
+```
+{% endtab %}
+{% tab 'Scala 3' %}
 ```scala
 // a type class
 trait Showable[A]:
   extension(a: A) def show: String
 ```
+{% endtab %}
+{% endtabs %}
 
 This is the Scala 3 way of saying that any type that implements this trait must define how the `show` method works.
 Notice that the syntax is very close to a normal trait:
 
+{% tabs 'trait' class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+```scala
+// a trait
+trait Show {
+  def show: String
+}
+```
+{% endtab %}
+{% tab 'Scala 3' %}
 ```scala
 // a trait
 trait Show:
   def show: String
 ```
+{% endtab %}
+{% endtabs %}
 
 There are a few important things to point out:
 
 1. Type-classes like `Showable` take a type parameter `A` to say which type we provide the implementation of `show` for; in contrast, normal traits like `Show` do not.
 2. To add the show functionality to a certain type `A`, the normal trait requires that `A extends Show`, while for type-classes we require to have an implementation of `Showable[A]`.
-3. To allow the same method calling syntax in both `Showable` that mimics the one of `Show`, we define `Showable.show` as an extension method.
+3. In Scala 3, to allow the same method calling syntax in both `Showable` that mimics the one of `Show`, we define `Showable.show` as an extension method.
 
 ### Implement concrete instances
 
 The next step is to determine what classes in your application `Showable` should work for, and then implement that behavior for them.
 For instance, to implement `Showable` for this `Person` class:
 
+{% tabs 'person' %}
+{% tab 'Scala 2 and 3' %}
 ```scala
 case class Person(firstName: String, lastName: String)
 ```
+{% endtab %}
+{% endtabs %}
+
+you’ll define a `given` value for `Showable[Person]` in Scala 3, or an `implicit` value for `Showable[Person]` in Scala 2.
 
-you’ll define a `given` value for `Showable[Person]`.
 This code provides a concrete instance of `Showable` for the `Person` class:
```

Comment:
Then delete this line as a follow-up (to https://github.com/scala/docs.scala-lang/pull/2787/files#r1184709045)
```suggestion
```

Reply:
Done.